### ISSUE TITLE
Convert Json to CSV issue 2

### ISSUE THREAD

**Issue by Karthigeyan14** [Cinchoo/ChoETL#325]
I am facing the issue. The field looks like this:

**JSON**

"extension": [
{
"Test": "value",
"value": [
"0",
"1",
"2"
]
},
{
"Test1": "Test",
"value": "20"
},
{
"Test2": "Test",
"value": "35"
}
]
The values 20 and 35 are missing because in the first object, the value field is a list, while in the subsequent objects, value is not a list. This inconsistency causes the values to be overlooked.

**Comment by Karthigeyan14** [comment]
@Cinchoo  #324 Same issue, but I didn't get the solution. Can you suggest a solution to this issue?

**Comment by Cinchoo** [comment]
can't reproduce it. here is working sample

https://dotnetfiddle.net/ipX7aU

Output:
```
extension_0_Test,extension_0_value_0,extension_0_value_1,extension_0_value_2,extension_1_Test1,extension_1_value,extension_2_Test2,extension_2_value
value,0,1,2,Test,20,Test,35
```

**Comment by Karthigeyan14** [comment]
I just showed an example; the actual file is different. I tried what you mentioned, but the CSV file output writes all the data in a single line. Since the JSON file contains a lot of data, I need to write each entry separately one by one. I’ve attached the example file. Can you suggest how to fix this?

[Sample.json](https://github.com/user-attachments/files/17649125/Sample.json)

**Comment by Cinchoo** [comment]
cinchoo etl does the best way possible to convert json to csv. but json is structured file, csv is flat. in order to accomblish the conversion expected way, some custom work is needed.

Based on your input, i've compiled sample fiddle

https://dotnetfiddle.net/zW9iI2

Expected output will be:

```
testUrl,resourceType,managingtest_test,name_0_lastname,name_0_firstname,name_0_given,gender,birthDate,sata_YearUpdated,url,value_0,value_1,value_2
https://,test,test/1809,,,,male,1925-01-01,2024,www.com,test1,test2,test3
https://,test,test/1809,,,,male,1925-01-01,2024,www.1.com,test20,,
https://,test,test/1809,,,,male,1925-01-01,2024,www.2.com,test30,,
https://,test,test/1809,,,,male,1925-01-01,2024,www.3.com,test40,,
https://,test,test/1810,,,,male,1925-01-01,2024,www.1.com,test20,,
https://,test,test/1810,,,,male,1925-01-01,2024,www.2.com,test30,,
https://,test,test/1810,,,,male,1925-01-01,2024,www.com,test1,test2,test3
https://,test,test/1810,,,,male,1925-01-01,2024,www.3.com,test40,,
```

Hope this works.

**Comment by Karthigeyan14** [comment]
Yes @Cinchoo , it works for the Value field, but in the Name field, it's showing empty. I need all the data from the JSON file to be converted into CSV format.

**Comment by Cinchoo** [comment]
there was issue pulling those data. applied fix, released ChoETL.Core v1.2.1.68

Here is working fiddle

https://dotnetfiddle.net/Q4HZo3
